connect/models/product.py

```diff
@@ -28,15 +29,20 @@ class Item(BaseModel):
     global_id = None  # type: str
     mpn = None  # type: str
     old_quantity = None  # type: int
-    quantity = None  # type: int
+    quantity = None  # type: Union[int, str]
 
 
 class ItemSchema(BaseSchema):
     global_id = fields.Str()
     mpn = fields.Str()
     old_quantity = fields.Integer()
-    quantity = fields.Integer()
+    quantity = fields.Str()
 
     @post_load
     def make_object(self, data):
+        # If quantity string contains a number, convert to int
+        if 'quantity' in data:
+            quantity = data['quantity']
+            if quantity.isdigit() or (quantity.startswith('-') and quantity[1:].isdigit()):
+                data['quantity'] = int(quantity)
         return Item(**data)
```

Comment:
It is ambiguous. For the user to work with this field, you need to check.
I think you need to do `Int` or `None` (not a string)

Reply:
You're right, I'll change it. An integer value or `None` when it is unlimited.

Reply:
By checking the documentation [here](https://confluence.int.zone/display/CONNECT/Asset+Object), it says that `quantity` and `old_quantity` are numbers, but in the provided example actually strings are given, which seems to also be the case with a real API call. Since I did not find `'unlimited'` mentioned there, I wasn't sure if there could be other possible string value, so I took the approach of making sure that is convertible to int before attempting the int() call (it would have probably been better to just try and in case it raises a `ValueError`, just store `None)`. If you say that the string will either contain a number or `'unlimited'`, then I'll just check that case.

About storing `-1` when unlimited, I think that Python offers the cleanest syntax of an optional int, which allows for checks like `if item.quantity` instead of `if item.quantity > -1`.

Reply:
By checking the documentation [here](https://confluence.int.zone/display/CONNECT/Asset+Object), it says that `quantity` and `old_quantity` are numbers, but in the provided example actually strings are given, which seems to also be the case with a real API call. Since I did not find `'unlimited'` mentioned there, I wasn't sure if there could be other possible string value, so I took the approach of making sure that is convertible to int before attempting the int() call (it would have probably been better to just try and in case it raises a `ValueError`, just store `None)`. If you say that the string will either contain a number or `'unlimited'`, then I'll just check that case.

About storing `-1` when unlimited, I think that Python offers the cleanest syntax of an optional int, which allows for checks like `if item.quantity` instead of `if item.quantity > -1`.

Reply:
Rigth, `quantity` equals type `int` or `str = unlimited`


If a`quantity = 0` you checked `if item.quantity` - False, but `quantity = 0` not `unlimited`
`None != unlimited`

Reply:
Sorry, I meant `if item.quantity is not None` instead of `if item.quantity > -1`. Looks more expressive.